packages/smart/smart.go

```diff
@@ -616,6 +618,20 @@ func (sc *SmartContract) CallContract(flags int) (result string, err error) {
 	payWallet := &model.Key{}
 	sc.TxContract.Extend = sc.getExtend()
 
+	retResult := func() (string, error) {
+		if err != nil {
+			eText := err.Error()
+			if !strings.HasPrefix(eText, `{`) {
+				err = script.SetVMError(`panic`, eText)
+			}
+		}
+		return result, err
+	}
+	retError := func(errMessage string) (string, error) {
+		err = fmt.Errorf(errMessage)
```

Comment:
Looks very complicated, closure that calls closure, maybe it would be easier just set err to err message and call return retResult()?